<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>QCPRmoClass</class>
 <widget class="QMainWindow" name="QCPRmoClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1106</width>
    <height>729</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>QCPRmo</string>
  </property>
  <property name="windowIcon">
   <iconset resource="qcprmo.qrc">
    <normaloff>:/Resources/Radar.ico</normaloff>:/Resources/Radar.ico</iconset>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayoutMain">
    <property name="margin">
     <number>4</number>
    </property>
    <property name="spacing">
     <number>4</number>
    </property>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1106</width>
     <height>29</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpenM"/>
    <addaction name="actionSaveM"/>
    <addaction name="actionSaveAsM"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>View</string>
    </property>
    <addaction name="actionViewMessages"/>
    <addaction name="actionViewFIndicator"/>
    <addaction name="actionTraceList"/>
    <addaction name="actionViewTIndicator"/>
    <addaction name="actionViewSIndicator"/>
    <addaction name="actionControlPanelQML"/>
   </widget>
   <widget class="QMenu" name="menuAbout">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuView"/>
   <addaction name="menuAbout"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <property name="windowTitle">
    <string>Toolbar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionOpen"/>
   <addaction name="actionSave"/>
   <addaction name="separator"/>
   <addaction name="actionRule"/>
   <addaction name="actionClear"/>
   <addaction name="separator"/>
   <addaction name="actionProperties"/>
   <addaction name="separator"/>
   <addaction name="actionPause"/>
  </widget>
  <widget class="QDockWidget" name="dockWidgetQ">
   <property name="windowTitle">
    <string>Control panel</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContentsQ">
    <layout class="QGridLayout" name="gridLayout">
     <item row="0" column="0">
      <widget class="QLocalControlQMLWidget" name="qLocalControlQMLWidget">
       <property name="toolTip">
        <string>QML-based control panel</string>
       </property>
       <property name="whatsThis">
        <string>Modification of scan programs parameters.</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dockWidgetM">
   <property name="windowTitle">
    <string>Messages</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContentsM">
    <layout class="QGridLayout" name="gridLayoutM">
     <property name="margin">
      <number>4</number>
     </property>
     <property name="spacing">
      <number>4</number>
     </property>
     <item row="0" column="0">
      <layout class="QGridLayout" name="gridLayoutCons">
       <property name="spacing">
        <number>0</number>
       </property>
       <item row="0" column="0">
        <widget class="QToolButton" name="toolButtonClear">
         <property name="text">
          <string>...</string>
         </property>
         <property name="icon">
          <iconset resource="qcprmo.qrc">
           <normaloff>:/Resources/standardbutton-clear-32.png</normaloff>:/Resources/standardbutton-clear-32.png</iconset>
         </property>
         <property name="autoRaise">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QToolButton" name="toolButtonPause">
         <property name="text">
          <string>...</string>
         </property>
         <property name="icon">
          <iconset resource="qcprmo.qrc">
           <normaloff>:/Resources/player_pause.png</normaloff>:/Resources/player_pause.png</iconset>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <property name="autoRaise">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="0" column="2">
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </item>
     <item row="1" column="0">
      <widget class="QScrollArea" name="scrollAreaM">
       <property name="autoFillBackground">
        <bool>true</bool>
       </property>
       <property name="widgetResizable">
        <bool>true</bool>
       </property>
       <property name="alignment">
        <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
       </property>
       <widget class="QWidget" name="scrollAreaWidgetContentsM">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>0</y>
          <width>140</width>
          <height>292</height>
         </rect>
        </property>
        <layout class="QGridLayout" name="gridLayoutConsole">
         <property name="margin">
          <number>0</number>
         </property>
         <property name="spacing">
          <number>0</number>
         </property>
         <item row="0" column="0">
          <widget class="QConsoleWidget" name="qConsoleWidget">
           <property name="toolTip">
            <string>Console</string>
           </property>
           <property name="whatsThis">
            <string>Console</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dockWidgetF">
   <property name="windowTitle">
    <string>F-Indicator</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContentsF">
    <layout class="QGridLayout" name="gridLayout_3">
     <property name="margin">
      <number>0</number>
     </property>
     <item row="0" column="0">
      <widget class="QFIndicatorWidget" name="qFIndicatorWidget">
       <property name="toolTip">
        <string>F-Indicator</string>
       </property>
       <property name="whatsThis">
        <string>F-Indicator</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <action name="actionOpen">
   <property name="icon">
    <iconset resource="qcprmo.qrc">
     <normaloff>:/Resources/standardbutton-open-32.png</normaloff>:/Resources/standardbutton-open-32.png</iconset>
   </property>
   <property name="text">
    <string>Open</string>
   </property>
   <property name="menuId" stdset="0">
    <number>222</number>
   </property>
  </action>
  <action name="actionSave">
   <property name="icon">
    <iconset resource="qcprmo.qrc">
     <normaloff>:/Resources/standardbutton-save-32.png</normaloff>:/Resources/standardbutton-save-32.png</iconset>
   </property>
   <property name="text">
    <string>Save</string>
   </property>
   <property name="menuId" stdset="0">
    <number>223</number>
   </property>
  </action>
  <action name="actionProperties">
   <property name="icon">
    <iconset resource="qcprmo.qrc">
     <normaloff>:/Resources/Options.bmp</normaloff>:/Resources/Options.bmp</iconset>
   </property>
   <property name="text">
    <string>Properties</string>
   </property>
   <property name="menuId" stdset="0">
    <number>227</number>
   </property>
  </action>
  <action name="actionOpenM">
   <property name="icon">
    <iconset resource="qcprmo.qrc">
     <normaloff>:/Resources/standardbutton-open-32.png</normaloff>:/Resources/standardbutton-open-32.png</iconset>
   </property>
   <property name="text">
    <string>Open</string>
   </property>
   <property name="menuId" stdset="0">
    <number>222</number>
   </property>
  </action>
  <action name="actionSaveM">
   <property name="icon">
    <iconset resource="qcprmo.qrc">
     <normaloff>:/Resources/standardbutton-save-32.png</normaloff>:/Resources/standardbutton-save-32.png</iconset>
   </property>
   <property name="text">
    <string>Save</string>
   </property>
   <property name="menuId" stdset="0">
    <number>223</number>
   </property>
  </action>
  <action name="actionSaveAsM">
   <property name="icon">
    <iconset resource="qcprmo.qrc">
     <normaloff>:/Resources/standardbutton-save-32.png</normaloff>:/Resources/standardbutton-save-32.png</iconset>
   </property>
   <property name="text">
    <string>Save as ...</string>
   </property>
   <property name="menuId" stdset="0">
    <number>224</number>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
   <property name="menuId" stdset="0">
    <number>225</number>
   </property>
  </action>
  <action name="actionViewMessages">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Messages</string>
   </property>
   <property name="menuId" stdset="0">
    <number>251</number>
   </property>
  </action>
  <action name="actionViewFIndicator">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>F-Indicator</string>
   </property>
   <property name="menuId" stdset="0">
    <number>250</number>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
   <property name="menuId" stdset="0">
    <number>226</number>
   </property>
  </action>
  <action name="actionTraceList">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Trace list</string>
   </property>
   <property name="menuId" stdset="0">
    <number>252</number>
   </property>
  </action>
  <action name="actionClear">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="qcprmo.qrc">
     <normaloff>:/Resources/standardbutton-clear-32.png</normaloff>:/Resources/standardbutton-clear-32.png</iconset>
   </property>
   <property name="text">
    <string>Clear</string>
   </property>
   <property name="visible">
    <bool>false</bool>
   </property>
   <property name="menuId" stdset="0">
    <number>229</number>
   </property>
  </action>
  <action name="actionRule">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="qcprmo.qrc">
     <normaloff>:/Resources/page-setup-32.png</normaloff>:/Resources/page-setup-32.png</iconset>
   </property>
   <property name="text">
    <string>Rule</string>
   </property>
   <property name="visible">
    <bool>false</bool>
   </property>
   <property name="menuId" stdset="0">
    <number>228</number>
   </property>
  </action>
  <action name="actionViewTIndicator">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>T-Indicator</string>
   </property>
   <property name="menuId" stdset="0">
    <number>253</number>
   </property>
  </action>
  <action name="actionViewSIndicator">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>S-Indicator</string>
   </property>
   <property name="menuId" stdset="0">
    <number>254</number>
   </property>
  </action>
  <action name="actionControlPanelQML">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="qcprmo.qrc">
     <normaloff>:/Resources/HAMMER.ICO</normaloff>:/Resources/HAMMER.ICO</iconset>
   </property>
   <property name="text">
    <string>Control panel QML</string>
   </property>
   <property name="toolTip">
    <string>Control panel QML</string>
   </property>
   <property name="menuId" stdset="0">
    <number>249</number>
   </property>
  </action>
  <action name="actionPause">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="qcprmo.qrc">
     <normaloff>:/Resources/player_pause.png</normaloff>:/Resources/player_pause.png</iconset>
   </property>
   <property name="text">
    <string>pause</string>
   </property>
   <property name="toolTip">
    <string>Pause oscillograph</string>
   </property>
   <property name="visible">
    <bool>false</bool>
   </property>
   <property name="menuId" stdset="0">
    <number>230</number>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>QLocalControlQMLWidget</class>
   <extends>QWidget</extends>
   <header>qlocalcontrolqmlwidget.h</header>
  </customwidget>
  <customwidget>
   <class>QFIndicatorWidget</class>
   <extends>QWidget</extends>
   <header>qfindicatorwidget.h</header>
  </customwidget>
  <customwidget>
   <class>QConsoleWidget</class>
   <extends>QWidget</extends>
   <header>qconsolewidget.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="qcprmo.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>mainToolBar</sender>
   <signal>actionTriggered(QAction*)</signal>
   <receiver>QCPRmoClass</receiver>
   <slot>onActionTriggered(QAction*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>152</x>
     <y>42</y>
    </hint>
    <hint type="destinationlabel">
     <x>79</x>
     <y>128</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>menuBar</sender>
   <signal>triggered(QAction*)</signal>
   <receiver>QCPRmoClass</receiver>
   <slot>onActionTriggered(QAction*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>187</x>
     <y>11</y>
    </hint>
    <hint type="destinationlabel">
     <x>79</x>
     <y>266</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>toolButtonClear</sender>
   <signal>clicked()</signal>
   <receiver>QCPRmoClass</receiver>
   <slot>onConsoleClear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>15</x>
     <y>316</y>
    </hint>
    <hint type="destinationlabel">
     <x>552</x>
     <y>364</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>toolButtonPause</sender>
   <signal>clicked()</signal>
   <receiver>QCPRmoClass</receiver>
   <slot>onConsolePause()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>45</x>
     <y>316</y>
    </hint>
    <hint type="destinationlabel">
     <x>552</x>
     <y>364</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>dockWidgetQ</sender>
   <signal>visibilityChanged(bool)</signal>
   <receiver>QCPRmoClass</receiver>
   <slot>onDocPanelVisibilityChanged(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>74</x>
     <y>171</y>
    </hint>
    <hint type="destinationlabel">
     <x>552</x>
     <y>364</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>dockWidgetF</sender>
   <signal>visibilityChanged(bool)</signal>
   <receiver>QCPRmoClass</receiver>
   <slot>onDocPanelVisibilityChanged(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>74</x>
     <y>678</y>
    </hint>
    <hint type="destinationlabel">
     <x>552</x>
     <y>364</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>dockWidgetM</sender>
   <signal>visibilityChanged(bool)</signal>
   <receiver>QCPRmoClass</receiver>
   <slot>onDocPanelVisibilityChanged(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>74</x>
     <y>450</y>
    </hint>
    <hint type="destinationlabel">
     <x>552</x>
     <y>364</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>qFIndicatorWidget</sender>
   <signal>SendCommand(int,int,double,double)</signal>
   <receiver>QCPRmoClass</receiver>
   <slot>onSendRequest(int,int,double,double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>74</x>
     <y>690</y>
    </hint>
    <hint type="destinationlabel">
     <x>552</x>
     <y>364</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>qConsoleWidget</sender>
   <signal>SizeChanged(QSize)</signal>
   <receiver>QCPRmoClass</receiver>
   <slot>onConsoleSizeChanged(QSize)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>74</x>
     <y>480</y>
    </hint>
    <hint type="destinationlabel">
     <x>552</x>
     <y>364</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>qFIndicatorWidget</sender>
   <signal>SendFSelection(int,double,double,int,double,double,int)</signal>
   <receiver>QCPRmoClass</receiver>
   <slot>onSendFSelectionRequest(int,double,double,int,double,double,int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>74</x>
     <y>692</y>
    </hint>
    <hint type="destinationlabel">
     <x>552</x>
     <y>364</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>onActionTriggered(QAction*)</slot>
  <slot>onDocPanelVisibilityChanged(bool)</slot>
  <slot>onCheckConnection(bool*)</slot>
  <slot>onSendRequest(int,int,double,double)</slot>
  <slot>onConsoleClear()</slot>
  <slot>onConsoleSizeChanged(QSize)</slot>
  <slot>onConsolePause()</slot>
  <slot>onLocalControlCommand(char*,int,int)</slot>
  <slot>onGroundObjectAdded(double,double)</slot>
  <slot>onSendFSelectionRequest(int,double,double,int,double,double,int)</slot>
 </slots>
</ui>
